### PR TITLE
feat(event-manager): Adds implementation for ODPEventManager

### PR DESCRIPTION
**Pull request by yasirfolio3**
## Summary
This PR adds support for ODPEventManager. 

## Test plan
- Unit tests added

## Issues
- FSSDK-8515

### CODE REVIEW

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+func CompareSlices(a, b []string) bool {
+	// Required in case a nil value and an empty array is given
+	isFirstValueNil := a == nil
+	isSecondValueNil := b == nil
+
+	if !isFirstValueNil && !isSecondValueNil {
+		if len(a) != len(b) {
+			return false
+		}
+		for i, v := range a {
```
Comment (jaeopt) [inline pkg/utils/utils.go:30]: This will return false for same array values with different order. Wondering if we're ok for all use cases.

**Review by jaeopt — CHANGES_REQUESTED** [review]
Looks great! A few small changes suggested.

File: pkg/utils/utils.go (hunk 1 above)
Comment (yasirfolio3) [reply pkg/utils/utils.go:30]: good catch, fixing this.

**Review by jaeopt — APPROVED** [review]
LGTM

File: pkg/odp/event_api_manager.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		requester = utils.NewHTTPRequester(logging.GetLogger(sdkKey, "EventAPIManager"))
+	}
+	return &DefaultEventAPIManager{requester: requester}
+}
+
+// SendODPEvents sends events to ODP's RESTful API
+func (s *DefaultEventAPIManager) SendODPEvents(config Config, events []Event) (canRetry bool, err error) {
+
+	// Creating request
+	apiEndpoint := config.GetAPIHost() + eventsAPIEndpointPath
```
Comment (msohailhussain) [inline pkg/odp/event_api_manager.go:61]: use `fmt.Sprintf`

File: pkg/odp/event_api_manager.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	headers := []utils.Header{{Name: "Content-Type", Value: "application/json"}, {Name: ODPAPIKeyHeader, Value: config.GetAPIKey()}}
```
Comment (msohailhussain) [inline pkg/odp/event_api_manager.go:62]: add these as `const`

File: pkg/odp/event_manager.go [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+	if bm.eventAPIManager == nil {
+		bm.eventAPIManager = NewEventAPIManager(bm.sdkKey, nil)
+	}
+
+	return bm
+}
+
+// Start does not do any initialization, just starts the ticker
+func (bm *BatchEventManager) Start(ctx context.Context) {
+	if !bm.IsOdpServiceIntegrated() {
```
Comment (msohailhussain) [inline pkg/odp/event_manager.go:173]: I think this should be ready? please check in other sdks.

File: pkg/odp/segment_api_manager.go [hunk 5]
```diff
@@ -144,8 +138,8 @@ func (sm *DefaultSegmentAPIManager) FetchQualifiedSegments(config Config, userKe
 	requestQuery := sm.createRequestQuery(userKey, userValue, config.GetSegmentsToCheck())
 
 	// Creating request
-	apiEndpoint := config.GetAPIHost() + "/v3/graphql"
-	headers := []utils.Header{{Name: "Content-Type", Value: "application/json"}, {Name: "x-api-key", Value: config.GetAPIKey()}}
+	apiEndpoint := config.GetAPIHost() + graphqlAPIEndpointPath
```
Comment (msohailhussain) [inline pkg/odp/segment_api_manager.go:141]: `fmt`

File: pkg/odp/segment_api_manager_test.go [hunk 6]
```diff
@@ -306,28 +319,31 @@ func (s *SegmentAPIManagerTestSuite) TestCreateRequestQuery() {
...
 
 // Tests with live ODP server
-func (s *SegmentAPIManagerTestSuite) TestLiveOdpGraphQL() {
-	segmentsToCheck := []string{"segment-1"}
-	s.config = NewConfig(s.liveOdpAPIKey, s.liveOdpAPIHost, segmentsToCheck)
-	segments, err := s.segmentAPIManager.FetchQualifiedSegments(s.config, "fs_user_id", s.liveOdpValidUserID)
-	s.NoError(err)
-	s.Empty(segments, "none of the test segments in the live ODP server")
-}
+// func (s *SegmentAPIManagerTestSuite) TestLiveOdpGraphQL() {
```
Comment (msohailhussain) [inline pkg/odp/segment_api_manager_test.go:322]: is this intentional?

File: pkg/odp/segment_manager.go [hunk 7]
```diff
@@ -19,10 +19,11 @@ package odp
 
 import (
 	"fmt"
-
-	"github.com/optimizely/go-sdk/pkg/odp/utils"
 )
 
+// SMOptionConfig are the SegmentManager options that give you the ability to add one more more options before the segment manager is initialized.
+type SMOptionConfig func(em *DefaultSegmentManager)
```
Comment (msohailhussain) [inline pkg/odp/segment_manager.go:25]: shouldn't this be SMConfigOptions

File: pkg/utils/utils_test.go [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCompareSlices(t *testing.T) {
+	assert.True(t, CompareSlices(nil, nil))
+	assert.True(t, CompareSlices([]string{}, []string{}))
```
Comment (msohailhussain) [inline pkg/utils/utils_test.go:27]: can you try 
`make and []string{}`

File: pkg/odp/event_manager.go (hunk 4 above)
Comment (yasirfolio3) [reply pkg/odp/event_manager.go:173]: This is similar to `odpServiceIntegrated` property in `swift-sdk`

File: pkg/odp/segment_api_manager_test.go (hunk 6 above)
Comment (yasirfolio3) [reply pkg/odp/segment_api_manager_test.go:322]: Yes, have put these here for debug purposes similar to `swift-sdk`

File: pkg/odp/segment_manager.go (hunk 7 above)
Comment (yasirfolio3) [reply pkg/odp/segment_manager.go:25]: Changing it into `SMOptionFunc` which is in sync with the older options in the sdk.

**Pull request merged**